content/posts/2020-03-31-continuous-stuff-with-github-actions/index.mdx

```diff
@@ -0,0 +1,256 @@
+
+You could delegate everything into a ready-to-go Docker container, and run whatever you need in there. But then you lose
+access to the overall filesystem of your original job, where you already cloned your project and created a bunch of
+useful artifacts. Not great, either.
+
+GitHub Actions solves this by allowing you to write reusable actions in two different ways. None of those are YAML, and
+both of those are way better:
+
+1. As a docker container. But one that will automatically mount the local filesystem into it, so that you get access to
+   everything your job has been doing so far. And since, it's docker, you can pretty much install whatever dependencies
+   you need, without caring about polluting the host job
+2. In JavaScript. Which, as you may know, is a pretty powerful tool.
+
+Here's a basic GitHub Actions job:
+
+```yaml
+name: Demoo
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+  steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+
+    - name: Install deps
+      run: npm install
+
+    - name: Run tests
+      run: npm run test
+```
+
+The first step here, `actions/checkout@v2` is what checks out your repo to the local filesystem. And it's
+actually done in JavaScript! The syntax, as you may tell, looks suspiciously like a link to a GitHub repo. And, well, it is. If you check [release `v2` on that repository][actions-checkout], you'll see the actual JS code used to clone your repo into the action.
+
+And of course, you're free to fork this action and edit it with your own customizations, if you need. The exact same
+flow that GitHub already allows for regular open-source work, now applied to their own CI.
+
+## Ok, what's the catch?
+
+As you may be able to tell, I like GitHub Actions quite a lot. They're not without their shortcomings though.
+
+I have 2 major concerns:
+
+1. **Still a new kid in the block**. Public beta opened less than a year ago, which might be good enough of a reason to think twice before jumping into the hype-train. It's still in it's infancy, and as such, a lot of things may not be as polished as you'd expect. There's quite a lot of missing features being requested by the community, and hopefully we'll see them implemented soon enough.
+2. **No SSH access for debugging**. This might fall under the previous point. But the lack of ability to debug failed actions by
+   SSH'ing into the container was almost a deal breaker for me. I spent countless hours debugging things via
+   trial-and-error that would have been way faster had I just been able to just see what happened for myself
+
+## Wrapping up
+
+These drawbacks, as concerning as they may have been, didn't prevent me from using GitHub Actions over the past few
+months, and even performing a company-wide migration at [Utrust][utrust]. This post was my way of compiling a short
+tutorial around how I started to use them, and how they're so much different than previous CIs I've tried.
```

Comment:
As a conclusion, you're missing a review of the other points of the post. As it is, it feels like it should simply be the last paragraph of the previous section.

Reply:
true. added a bit more here